src/App.tsx

```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[],
+  selectedGoods: string[]
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    selectedGoods: ['Jam'],
+  };
+
+  getButtonAction = (action: string, good?: string) => {
+    if (good && action === 'select') {
+      this.setState((state) => ({ selectedGoods: [...state.selectedGoods, good] }));
+    } else if (good && action === 'remove') {
+      this.setState((state) => ({ selectedGoods: state.selectedGoods.filter(el => el !== good) }));
+    } else {
+      this.setState({ selectedGoods: [] });
+    }
+  };
+
+  render() {
+    const { selectedGoods, goods } = this.state;
+
+    return (
+      <div className="app">
+        <h1 className="title">
+          {selectedGoods.length > 0
+            ? `${selectedGoods.join(', ')} ${selectedGoods.length > 1 ? 'are' : 'is'}`
+              + ' selected'
```

Comment:
try to avoid so long lines that you should break, it's difficult to read, 
if it's impossible to place it one line, you can create a method that create a needed message